From 7474da4cb83e6e87b7c7eb2601b9368806f23c42 Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Fri, 8 Sep 2017 10:52:58 +0200
Subject: [PATCH] osutil: adjust StreamCommand tests for golang 1.9

In golang 1.9 there are richer error constructs returned from certain
operations and tests were very precisely monitoring the result. This
patch adjust tests to work on both golang 1.9 and earlier.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 osutil/exec_test.go | 10 ++++++----
 osutil/io_test.go   |  3 ++-
 2 files changed, 8 insertions(+), 5 deletions(-)

--- a/osutil/exec_test.go
+++ b/osutil/exec_test.go
@@ -207,8 +207,9 @@
 
 	wrf, wrc := osutil.WaitingReaderGuts(stdout)
 	c.Assert(wrf, FitsTypeOf, &os.File{})
-	c.Check(wrf.(*os.File).Close(), Equals, syscall.EINVAL) // i.e. already closed
-	c.Check(wrc.ProcessState, NotNil)                       // i.e. already waited for
+	// Depending on golang version the error is one of the two.
+	c.Check(wrf.(*os.File).Close(), ErrorMatches, "invalid argument|file already closed")
+	c.Check(wrc.ProcessState, NotNil) // i.e. already waited for
 }
 
 func (s *execSuite) TestStreamCommandSad(c *C) {
@@ -221,6 +222,7 @@
 
 	wrf, wrc := osutil.WaitingReaderGuts(stdout)
 	c.Assert(wrf, FitsTypeOf, &os.File{})
-	c.Check(wrf.(*os.File).Close(), Equals, syscall.EINVAL) // i.e. already closed
-	c.Check(wrc.ProcessState, NotNil)                       // i.e. already waited for
+	// Depending on golang version the error is one of the two.
+	c.Check(wrf.(*os.File).Close(), ErrorMatches, "invalid argument|file already closed")
+	c.Check(wrc.ProcessState, NotNil) // i.e. already waited for
 }
